client/ayon_harmony/api/plugin.py

```diff
@@ -1,23 +1,159 @@
+
+            shared_data["harmony_cached_scene_data"] = scene_data
+            shared_data["harmony_cached_instance_data"] = cache
+            shared_data["harmony_cached_legacy_instances_names"] = cache_legacy
+        return shared_data
+
+
+class HarmonyCreator(Creator, HarmonyCreatorBase):
     """Creator plugin to create instances in Harmony.
 
     By default a Composite node is created to support any number of nodes in
     an instance, but any node type is supported.
     If the selection is used, the selected nodes will be connected to the
     created node.
     """
-    # TODO: Refactor to the new creator API
 
-    defaults = ["Main"]
-    node_type = "COMPOSITE"  # TODO remove this
     settings_category = "harmony"
 
-    auto_connect = False
-    composition_node_pattern = ""
+
+    def create(self, product_name, instance_data, pre_create_data):
+        # Create the node
+        node = self.product_impl(product_name, instance_data, pre_create_data)
+
+        instance = CreatedInstance(
+            self.product_type,
+            product_name,
+            instance_data,
+            self
+        )
+        instance.transient_data["node"] = node
+        harmony.imprint(node, instance.data_to_store())
+
+        self._add_instance_to_context(instance)
+
+    def update_instances(self, update_list):
+        for created_inst, _changes in update_list:
+            node = created_inst.transient_data["node"]
+            new_data = created_inst.data_to_store()
+
+            # Use the node's active state to store the instance's active state
+            active = new_data.pop("active", True)
+            harmony.send(
+                {"function": "AyonHarmonyAPI.setState",
+                 "args": [[node], [active]]}
+            )
+
+            harmony.imprint(node, new_data)
+
+    def remove_instances(self, instances):
+        for instance in instances:
+            # There is only ever one workfile instance
+            harmony.delete_node(instance.transient_data["node"])
+            self._remove_instance_from_context(instance)
+
+    def collect_instances(self):
+        cache = self.cache_instance_data(self.collection_shared_data)
+        for node_name in cache.get("harmony_cached_instance_data").get(
+                self.identifier, []):
+            data = cache.get("harmony_cached_scene_data")[node_name]
+
+            product_type = data.get("productType")
+            if product_type is None:
+                product_type = data["family"]
+                data["productType"] = product_type
+            data["family"] = product_type
+
+            instance = CreatedInstance.from_existing(instance_data=data,
+                                                     creator=self)
+            instance.transient_data["node"] = node_name
+
+            # Active state is based of the node's active state
+            instance.data["active"] = harmony.send(
```

Comment:
Got a small issue here: Template instance is always disabled by default, because it is a Backdrop and the active attribute considers it should be a node, and note.getEnable() returns false when the node is disabled OR if not found... Which is a strange API implementation but we must deal with it by testing template as backdrops, and maybe consider they are enabled by default.

Reply:
Enhanced `isEnabled` and `setState` to take `Backdrop` into account. 
To learn state of Backdrop only first node inside is checked.
If `Backdrop` instance is disabled, all their nodes are disabled.